source/develop/release-management/features/storage/discard-after-delete.html.md

```diff
@@ -0,0 +1,100 @@
+
+This feature is about freeing (discarding) the whole disk space right before removing it.
+
+### For whom is this feature useful
+This feature is useful for users that have thinly provisioned luns and wish to reclaim space when removing a disk from a block storage domain that is built from them.
+
+
+## Owner
+* Idan Shaby
+* Email: <ishaby@redhat.com>
+
+
+## Current status
+* [Phase one](#usage) (implemented in vdsm config file) - released in oVirt 4.0.
+* [Phase two](#future-plans) (configurable on engine side) - planned for oVirt 4.1.
+
+
+## General functionality and restrictions
+
+### Under the hood
+A new parameter was added to the *vdsm* configuration file, ***discard_enable***.<br/>
+When it is *true* in a specific vdsm host, that host will issue a *blkdiscard* command on a logical volume right before removing it. That means that each disk or snapshot that this host removes, automatically triggers  it to inform the storage array that this logical volume's blocks were freed and can be used by other luns in the array.
+
+### Possible consumers of the discarded blocks
+After a disk is removed from a block storage domain and its blocks are discarded, there's no restriction to use it. That includes:
+
+* New and existing disks on the same storage domain (if there's enough free space in the storage domain, of course).
+* New and existing disks on other storage domains that are built on luns from the same storage array.
+* Other processes that use luns from the same storage array.
+
+### Restrictions
+
+#### The underlying storage
+Let *lunX* be a device and *dm-X* be its corresponding dm device for a natural number X. Then *lunX* is considered to ***support discard*** iff the value of `/sys/block/dm-X/queue/discard_max_bytes` is bigger than 0 [^1].
+
+The lun in the underlying storage must *support discard* in order for this to work.<br/>
+If it does not, the *blkdiscard* command that *vdsm* tries to execute will fail with a log but will not break the lv removal. That is, the lv will be removed but the space will not be reclaimed.
+
+#### *Wipe After Delete* and *discard_enable*
+These two properties are completely independent.<br/>
+To understand it better, let's take a look at the flow of removing a disk from a block domain in *vdsm*:
+
+1. If the disk's *Wipe After Delete* is enabled, then the disk is first wiped. If not, *vdsm* moves to phase 2.
+2. If the host that is going to remove it has its *discard_enable* enabled, then *vdsm* issues a *blkdiscard* command on that disk's logical volumes. If not, it moves to phase 3.
+3. The disk's logical volumes are removed.
+
+
+## Usage
+To make it work, the administrator should set the *discard_enable* property in *vdsm*'s `config.py.in` file to ***true*** and then **restart the *vdsmd* service**.
```

Comment:
Actually it's non of them, thanks for noticing.
Users should use the /etc/vdsm/vdsm.conf file.